components/script/dom/node.rs

```diff
@@ -951,16 +950,17 @@ impl Node {
         if let Some(ref shadow_root) = self.downcast::<ShadowRoot>() {
```

Comment:
`owner_shadow_root` -> `containing_shadow_root`? And why is this special-case needed? The containing shadow root's rare data pointer should point to itself, then you don't need it.

Otherwise you also need it on the layout accessors.